Sources/client-sdk-swift/momento/TopicClient.swift

```diff
@@ -11,19 +15,52 @@ public class TopicClient: TopicClientProtocol {
     var credentialProvider: CredentialProviderProtocol
     
     @available(macOS 10.15, *)
-    init(configuration: TopicClientConfiguration, credentialProvider: CredentialProviderProtocol, logger: MomentoLoggerProtocol = DefaultMomentoLogger(loggerName: "swift-momento-logger", level: DefaultMomentoLoggerLevel.info)) {
+    init(
+        configuration: TopicClientConfiguration,
+        credentialProvider: CredentialProviderProtocol,
+        logger: MomentoLoggerProtocol = DefaultMomentoLogger(
+            loggerName: "swift-momento-logger",
+            level: DefaultMomentoLoggerLevel.info
+        )
+    ) {
         self.logger = logger
         self.credentialProvider = credentialProvider
-        self.pubsubClient = PubsubClient(logger: logger, configuration: configuration, credentialProvider: credentialProvider)
+        self.pubsubClient = PubsubClient(
+            logger: logger,
+            configuration: configuration,
+            credentialProvider: credentialProvider
+        )
     }
     
-    public func publish() async -> String {
+    public func publish(
+        cacheName: String,
+        topicName: String,
+        value: String
+    ) async -> TopicPublishResponse {
+        if cacheName.count < 1 {
+            return TopicPublishError(
+                error: SdkError(
+                    message: "Must provide a cache name",
+                    errorCode: MomentoErrorCode.INVALID_ARGUMENT_ERROR)
+            )
+        }
+        if topicName.count < 1 {
+            return TopicPublishError(
+                error: SdkError(
+                    message: "Must provide a topic name",
+                    errorCode: MomentoErrorCode.INVALID_ARGUMENT_ERROR)
+            )
+        }
         do {
-            let result = try await self.pubsubClient.publish()
+            let result = try await self.pubsubClient.publish(
+                cacheName: cacheName,
+                topicName: topicName,
+                value: value
+            )
+            return result
         } catch {
-            print("TopicClientPublishError: \(error)")
+            return error as! TopicPublishResponse
```

Comment:
```suggestion
            return TopicPublishError(error: UnknownError(message: "Unknown error from publish", innerException: error))
```
This needs to be returned as `TopicPublishError` so it can be pattern matched as an error response.